spec/services/gpo_confirmation_maker_spec.rb

```diff
@@ -72,30 +74,39 @@
     end
   end
 
-  context 'with a nil zipcode' do
-    let(:zipcode) { nil }
-
-    describe '#perform' do
-      it 'accepts the zipcode' do
-        subject.perform
-
-        gpo_confirmation = GpoConfirmation.first
-        entry_hash = gpo_confirmation.entry
-        expect(entry_hash[:zipcode]).to be_nil
-      end
-    end
-  end
-
-  context 'with a (bogus) zip+1 zipcode' do
-    let(:zipcode) { '12345+0' }
-
-    describe '#perform' do
-      it 'strips the +0 from the zipcode' do
-        subject.perform
-
-        gpo_confirmation = GpoConfirmation.first
-        entry_hash = gpo_confirmation.entry
-        expect(entry_hash[:zipcode]).to eq '12345'
+  [
+    [nil, false],
+    ['1234', false],
+    ['12345+0', '12345'],
+    ['12345 - 0', '12345'],
+    ['12345- 0', '12345'],
+    ['12345-0', '12345'],
+    ['12345-6', '12345'],
+    ['12345-67', '12345'],
+    ['12345-678', '12345'],
+    ['12345-6789', '12345-6789'],
```

Comment:
Worth adding one with too many digits? e.g. `['12345-67890', '12345-6789']`

Reply:
Yeah I added one. Honestly I'm not sure what the best choice is here. The spirit is to try and pull a 5 digit zip code out even if the rest is garbage. A stricter validator would reject all but 5 and 9 digit variants.